app/src/main/java/com/tari/android/wallet/ffi/FFICompletedTx.kt

```diff
@@ -91,79 +91,79 @@ internal class FFICompletedTx constructor(pointer: FFICompletedTxPtr): FFIBase()
     fun getId(): BigInteger {
         val error = FFIError()
         val bytes = jniGetId(ptr, error)
-        if (error.code != 0) {
-            throw RuntimeException()
+        if (error.code != NO_ERROR.code) {
+            throw FFIException(error)
         }
         return BigInteger(1, bytes)
     }
 
     fun getDestinationPublicKey(): FFIPublicKey {
         val error = FFIError()
         val result = FFIPublicKey(jniGetDestinationPublicKey(ptr, error))
-        if (error.code != 0) {
-            throw RuntimeException()
+        if (error.code != NO_ERROR.code) {
+            throw FFIException(error)
         }
         return result
     }
 
     fun getSourcePublicKey(): FFIPublicKey {
         val error = FFIError()
         val result = FFIPublicKey(jniGetSourcePublicKey(ptr, error))
-        if (error.code != 0) {
-            throw RuntimeException()
+        if (error.code != NO_ERROR.code) {
+            throw FFIException(error)
         }
         return result
     }
 
     fun getAmount(): BigInteger {
         val error = FFIError()
         val bytes = jniGetAmount(ptr, error)
-        if (error.code != 0) {
-            throw RuntimeException()
+        if (error.code != NO_ERROR.code) {
+            throw FFIException(error)
         }
         return BigInteger(1, bytes)
     }
 
     fun getFee(): BigInteger {
         val error = FFIError()
         val bytes = jniGetFee(ptr, error)
-        if (error.code != 0) {
-            throw RuntimeException()
+        if (error.code != NO_ERROR.code) {
+            throw FFIException(error)
         }
         return BigInteger(1, bytes)
     }
 
     fun getTimestamp(): BigInteger {
         val error = FFIError()
         val bytes = jniGetTimestamp(ptr, error)
-        if (error.code != 0) {
-            throw RuntimeException()
+        if (error.code != NO_ERROR.code) {
+            throw FFIException(error)
         }
         return BigInteger(1, bytes)
     }
 
     fun getMessage(): String {
         val error = FFIError()
         val result = jniGetMessage(ptr, error)
-        if (error.code != 0) {
-            throw RuntimeException()
+        if (error.code != NO_ERROR.code) {
+            throw FFIException(error)
         }
         return result
     }
 
     fun getStatus(): Status {
         val error = FFIError()
         val status = jniGetStatus(ptr, error)
-        if (error.code != 0) {
-            throw RuntimeException()
+        if (error.code != NO_ERROR.code) {
+            throw FFIException(error)
         }
         return when (status) {
             -1 -> Status.TX_NULL_ERROR
             0 -> Status.COMPLETED
             1 -> Status.BROADCAST
             2 -> Status.MINED
             3 -> Status.UNKNOWN
-            else -> throw InvalidObjectException("Unexpected status")
+            else -> throw FFIException(message = "Unknown status: $status")
```

Comment:
```
else -> throw FFIException(message = "Undefined status: $status")
```
Unknown is present on the line above.

Reply:
Else is there to notify any unexpected status from the lib. "Unexpected status" is a better description actually, fixing 👍